scipy/integrate/_quadrature.py

```diff
@@ -472,8 +475,8 @@ def cumulative_trapezoid(y, x=None, dx=1.0, axis=-1, initial=None):
     res = np.cumsum(d * (y[slice1] + y[slice2]) / 2.0, axis=axis)
 
     if initial is not None:
-        if not np.isscalar(initial):
-            raise ValueError("`initial` parameter should be a scalar.")
+        if initial is not 0:
```

Comment:
```suggestion
        if initial != 0:
```
This is what's causing the test failures.